### ISSUE TITLE
Re-onboarding after disabling Dev Mode

### ISSUE THREAD

**Issue by thenbrent** [Automattic/woocommerce-payments#666]
A question that came up in a [recent review](https://github.com/Automattic/woocommerce-payments/pull/660#discussion_r426795983): _can a store re-onboard without issues after disabling Dev Mode_?

Right now, in the readme and [Dev Mode doc](https://docs.woocommerce.com/document/payments/testing/dev-mode/), we recommend only using Dev Mode on a store _that will **never need to process real payments**_.

If a store can onboard a live account after disabling Dev Mode without any negative side effects, we could expand that recommendation.

**Comment by LevinMedia** [comment]
@dechov @thenbrent Would it be possible to have a "Delete account" feature in dev mode, so that when a developer is done testing, they could destroy their test account and have a merchant re-onboard using their own details?

**Comment by thenbrent** [comment]
Great discussion related to this here: p1643413022442609-slack-CGGCLBN58

**Comment by daquinons** [comment]
> Haven't tested this myself, but the "dev to live" case is currently coded as an explicit exception to the rule that prevents re-onboarding, as described in #301.

According to this, it should already be possible to reonboard from dev to live. This issue is about testing that case, and if it's working without any additional change, to modify [the doc](https://docs.woocommerce.com/document/payments/testing/dev-mode/) to make this possibility explicit in them with the help of Payments Ops

**Comment by brandonwanA8C** [comment]
I don't have or know the original "requirement for Dev Mode".
But from a Product's perspective, the "Dev Mode" use-case is to allow a user (eg. Builder) to TEST the end-to-end Store to Payments experience without having to use Real KYC info or Real money.

**Comment by dmallory42** [comment]
I did some testing on this and I can confirm the current behaviour is as follows:

- After onboarding a new account in dev mode (this used test mode in Stripe Onboarding, as we'd expect), I turned dev mode off.
- When I went back to WCPay, I was taken back to the onboarding screen. 
- This time, when I tried to onboard I needed to use real account details. But everything worked fine.

This seems to be the expected/desired behaviour. Do we think we need to make any modification to the documentation? It does seem to have a [section](https://woocommerce.com/document/woocommerce-payments/testing-and-troubleshooting/testing/dev-mode/#disabling-dev-mode) which discusses this case although perhaps we could make it a bit more transparent.
